src/steamship/cli/local_server/handler.py

```diff
@@ -0,0 +1,311 @@
+import json
+import logging
+import re
+import sys
+import uuid
+from http import HTTPStatus, server
+from typing import Callable, Dict, Optional, Type
+from urllib.parse import parse_qs, urlparse
+
+from steamship import Configuration, MimeTypes, Steamship, SteamshipError
+from steamship.data.user import User
+from steamship.data.workspace import SignedUrl
+from steamship.invocable import (
+    Invocable,
+    InvocableRequest,
+    InvocableResponse,
+    Invocation,
+    InvocationContext,
+    LoggingConfig,
+    PackageService,
+)
+from steamship.utils.signed_urls import upload_to_signed_url
+
+
+def internal_handler(  # noqa: C901
+    invocable_cls_func: Callable[[], Type[Invocable]],
+    event: Dict,
+    client: Steamship,
+    invocation_context: InvocationContext,
+) -> InvocableResponse:
+    invocable_class: Type[Invocable] = invocable_cls_func()
+
+    try:
+        request = InvocableRequest.parse_obj(event)
+    except SteamshipError as se:
+        logging.exception(se)
+        return InvocableResponse.from_obj(se)
+    except Exception as ex:
+        logging.exception(ex)
+        return InvocableResponse.error(
+            code=HTTPStatus.INTERNAL_SERVER_ERROR,
+            message="Plugin/App handler was unable to parse inbound request.",
+            exception=ex,
+        )
+
+    if request and request.invocation:
+        error_prefix = (
+            f"[ERROR - {request.invocation.http_verb} {request.invocation.invocation_path}] "
+        )
+    else:
+        error_prefix = "[ERROR - ?VERB ?PATH] "
+
+    if request.invocation.invocation_path == "/__dir__":
+        # Return the DIR result without (1) Constructing invocable_cls or (2) Parsing its config (in the constructor)
+        try:
+            return InvocableResponse(json=invocable_class.__steamship_dir__(invocable_class))
+        except SteamshipError as se:
+            logging.exception(se)
+            return InvocableResponse.from_obj(se)
+        except Exception as ex:
+            logging.exception(ex)
+            return InvocableResponse.error(
+                code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                prefix=error_prefix,
+                message="Unable to initialize package/plugin.",
+                exception=ex,
+            )
+
+    try:
+        invocable = invocable_cls_func()(
+            client=client, config=request.invocation.config, context=invocation_context
+        )
+        # TODO: We don't want to run this every time, but for now we are.
+        logging.info("Running __instance_init__")
+        invocable.instance_init()
```

Comment:
Is this the only difference from the version of this method in `lambda_handler`?  If so, would it not be better to just call the instance init after you've gotten the handler so that you could just use `lambda_handler.py`'s version of this method and not have to copy/paste the rest? The lambda_handler code is a little confusing / ugly already, so avoiding replicating it seems worthwhile if possible.